### ISSUE TITLE
Unable to cast from uchar* to float4*.

### ISSUE THREAD

**Issue by Masa-tam** [google/clspv#729]
When the following code is compiled:

```
__kernel void castErr(__global uchar *pOut)
{
	__global float4* pTmp = (__global float4*) pOut;
	pTmp->w = 0;
}
```

`UNREACHABLE execute at D:\src\clspv\lib\ReplacePointerBitcastPass.cpp:867!`

It will output the above error message and exit.

**Comment by Masa-tam** [comment]
The environment in which this occurs is as follows

Source : SHA-1: af397a5ac19fc11afd879e52aeaa4edd21f2b4fd
OS : Windows 10 Professional x64 20H2 (Build 19042.867)
Compiler : Visual Studio 2019 Version 16.9.3

**Comment by rjodinchr** [comment]
This is fixed with #846 (might already be the case in `main`)
